### DISCUSSION TITLE
VulkanSceneGraph-1.1.4, vsgPoints-0.5.0 and vsgExamples-1.1.4 development versions released

### DISCUSSION THREAD

**Discussion by vsg-dev** [vsg-dev/VulkanSceneGraph#1197] [Announcements]
Hi All,

I have tagged development releases for VulkanSceneGraph, vsgPoints and VsgExamples:

* [Changes in VulkanSceneGraph-1.1.4](https://github.com/vsg-dev/VulkanSceneGraph/releases/tag/v1.1.4)
* [Changes in vsgPoints-0.5.0](https://github.com/vsg-dev/vsgPoints/releases/tag/v0.5.0)
* [Changes in vsgExamples-1.1.4](https://github.com/vsg-dev/vsgExamples/releases/tag/v1.1.4)

There is nothing major in these releases, mostly minor bug fixes, API refinements, updates to build cleanly on g++13 (it's now standard on Ubuntu 24.04 and Kubuntu 24.04) and to provide some refinements that make vsg::Builder more extensible.

In terms of functionality the 1.1.x development series is closing in on what will be in the VulkanSceneGraph-1.2 stable release.  My focus now is on code and build refinements and bug fixing.  As we're getting close to a stable release I would appreciate users updating to VSG master or the 1.1.4 release and testing it against their applications.  If there are outstanding issues let me know.

Cheers,
Robert.

**Comment by VasanthBalguri** [comment]
Hi Robert,

Will 1.2 be long term stable release? Like when will the API be frozen?

Thanks and regards,
Vasanth

**Comment by robertosfield** [comment]
The approach I will aim for it to create a VulkanSceneGraph-1.2 branch and make the initial 1.2.0 stable release from this.  If/when bugs are reported then fixed, ones that can be applied to the VulkanSceneGraph-1.2 branch will be, then when it looks like it's required I'll make a 1.2.1 release, and so on.

My aim will be for the 1.2.x stable series to be ABI compatible, but this isn't always possible, so API compatible will be the next goal.  My experience with the OpenSceneGraph project is that it's not always possible to keep compatibility as parts of the API might be the cause of bug/build issues.  The more testing we get prior to make the stable branch the better shape we'll be in for the stable release series.

One the VulkanSceneGraph-1.2 branch is made it'll live it;s life, and VulkanSceneGraph master will continue to evolve, a future 1.3.x developer release series will be made from this, and when things converge to another stable point I'll make VulkanSceneGraph-1.4 branch and start the 1.4.x stable release series.  

The work on master after 1.2 branch will loosely aim to maintain API compatibility when appropriate but won't be constrained by it - if an API change is required to improve the VSG then I'll allow it to be merged in and inform the community that they should be aware of a breaking API change.  The VSG's compile version system is the way for users to manage any of these changes.

The goal is to make the VSG the best scene graph for as many different use case as we can, achieving this will occasionally create tension between maintaining API compatibility and making improvements.  I know the pain that API changes can cause so it's not something changed on a whim, if a breaking change is made it'll be because it offers a real long term improvement.
